Index: b/intl/locale.alias
===================================================================
--- a/intl/locale.alias	2008-08-08 19:09:00.000000000 +0200
+++ b/intl/locale.alias	2008-08-08 19:09:00.000000000 +0200
@@ -34,8 +34,8 @@
 dansk		da_DK.ISO-8859-1
 deutsch		de_DE.ISO-8859-1
 dutch		nl_NL.ISO-8859-1
-eesti		et_EE.ISO-8859-1
-estonian	et_EE.ISO-8859-1
+eesti		et_EE.ISO-8859-15
+estonian	et_EE.ISO-8859-15
 finnish         fi_FI.ISO-8859-1
 français	fr_FR.ISO-8859-1
 french		fr_FR.ISO-8859-1
