--- a/README	2007-10-18 18:54:40.000000000 +0200
+++ b/README	2009-02-27 20:34:27.000000000 +0100
@@ -26,7 +26,7 @@
 
 ---
 
-This directory contains the version 2.7 release of the GNU C Library.
+This directory contains the version 2.9 release of the GNU C Library.
 
 The GNU C Library is the standard system C library for all GNU systems,
 and is an important part of what makes up a GNU system.  It provides the
