--- libc/ports/sysdeps/unix/sysv/linux/hppa/linuxthreads/bits/pthreadtypes.h	2007-05-01 21:15:38.000000000 +020
+++ libc/ports/sysdeps/unix/sysv/linux/hppa/linuxthreads/bits/pthreadtypes.h	2007-05-01 21:16:03.000000000 +0200
@@ -134,7 +134,7 @@
 } pthread_rwlockattr_t;
 #endif
 
-#ifdef __USE_XOPEN2K
+#if defined __USE_UNIX98 || defined __USE_XOPEN2K
 /* POSIX spinlock data type.  */
 typedef __atomic_lock_t pthread_spinlock_t;
 
===================================================================
RCS file: /cvs/glibc/ports/sysdeps/unix/sysv/linux/hppa/linuxthreads/bits/pthreadtypes.h,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- ports/sysdeps/unix/sysv/linux/hppa/linuxthreads/bits/pthreadtypes.h	2006/04/19 05:35:34	1.1
+++ ports/sysdeps/unix/sysv/linux/hppa/linuxthreads/bits/pthreadtypes.h	2007/05/02 02:07:38	1.2
@@ -111,8 +111,7 @@
 /* Once-only execution */
 typedef int pthread_once_t;
 
-
-#ifdef __USE_UNIX98
+#if defined __USE_UNIX98 || defined __USE_XOPEN2K
 /* Read-write locks.  */
 typedef struct _pthread_rwlock_t
 {
