Index: eglibc-2.18/elf/Makefile
===================================================================
--- eglibc-2.18.orig/elf/Makefile	2014-02-25 20:47:20.859469989 -0500
+++ eglibc-2.18/elf/Makefile	2014-02-25 20:47:20.769469646 -0500
@@ -424,7 +424,7 @@
 		    -D'SLIBDIR="$(slibdir)"' -DIS_IN_ldconfig=1
 CFLAGS-dl-cache.c = $(SYSCONF-FLAGS)
 CFLAGS-cache.c = $(SYSCONF-FLAGS)
-CFLAGS-rtld.c = $(SYSCONF-FLAGS)
+CFLAGS-rtld.c += $(SYSCONF-FLAGS)
 
 CPPFLAGS-.os += $(if $(filter $(@F),$(patsubst %,%.os,$(all-rtld-routines))),\
 		     -DNOT_IN_libc=1 -DIS_IN_rtld=1 -DIN_LIB=rtld)
