--- libc/ports/sysdeps/unix/sysv/linux/hppa/linuxthreads/bits/pthreadtypes.h	2007-05-01 21:15:38.000000000 +020
+++ libc/ports/sysdeps/unix/sysv/linux/hppa/linuxthreads/bits/pthreadtypes.h	2007-05-01 21:16:03.000000000 +0200
@@ -134,7 +134,7 @@
 } pthread_rwlockattr_t;
 #endif
 
-#ifdef __USE_XOPEN2K
+#if defined __USE_UNIX98 || defined __USE_XOPEN2K
 /* POSIX spinlock data type.  */
 typedef __atomic_lock_t pthread_spinlock_t;
 
