Index: glibc-2.22/sysdeps/unix/sysv/linux/hppa/setcontext.S
===================================================================
--- glibc-2.22.orig/sysdeps/unix/sysv/linux/hppa/setcontext.S
+++ glibc-2.22/sysdeps/unix/sysv/linux/hppa/setcontext.S
@@ -141,7 +141,7 @@ ENTRY(__setcontext)
 
 	/* No further context available. Exit now.  */
 	bl	HIDDEN_JUMPTARGET(exit), %r2
-	ldi	-1, %r26
+	ldi	0, %r26
 
 
 .Lerror:
