--- nptl/sysdeps/sparc/tls.h.orig	2007-07-26 00:36:08.000000000 +0200
+++ nptl/sysdeps/sparc/tls.h	2007-07-26 00:36:25.000000000 +0200
@@ -46,9 +46,15 @@
   dtv_t *dtv;
   void *self;
   int multiple_threads;
+#if __WORDSIZE == 64
+  int gscope_flag;
+#endif
   uintptr_t sysinfo;
   uintptr_t stack_guard;
   uintptr_t pointer_guard;
+#if __WORDSIZE != 64
+  int gscope_flag;
+#endif
 } tcbhead_t;
 
 #else /* __ASSEMBLER__ */

