commit e76c47eeb75382c25d232e97a2eba88b5ae2c046
Author: Ulrich Drepper <drepper@redhat.com>
Date:   Sat Mar 8 21:31:55 2008 +0000

    (____STRTOF_INTERNAL): Consume closing brace on NAN(...) sequence.

--- glibc-2.7.orig/stdlib/strtod_l.c
+++ glibc-2.7/stdlib/strtod_l.c
@@ -610,6 +610,9 @@
 		  mant = STRTOULL (startp + 1, &endp, 0);
 		  if (endp == cp)
 		    SET_MANTISSA (retval, mant);
+
+		  /* Consume the closing brace.  */
+		  ++cp;
 		}
 	    }
 
