--- a/elf/dl-lookup.c.original	2011-02-11 11:01:37.000000000 +0000
+++ b/elf/dl-lookup.c	2011-02-11 11:02:19.000000000 +0000
@@ -423,7 +423,7 @@ do_lookup_x (const char *undef_name, uin
 		     hash table.  */
 		  if (__builtin_expect (tab->size, 0))
 		    {
-		      assert (GLRO(dl_debug_mask) & DL_DEBUG_PRELINK);
+		      assert (GLRO_dl_debug_mask & DL_DEBUG_PRELINK);
 		      __rtld_lock_unlock_recursive (tab->lock);
 		      goto success;
 		    }
