Index: glibc-2.6.1/posix/glob.c
===================================================================
--- glibc-2.6.1.orig/posix/glob.c	2007-10-14 14:43:40.000000000 -0400
+++ glibc-2.6.1/posix/glob.c	2007-10-14 14:43:46.000000000 -0400
@@ -454,9 +454,11 @@
 	  dirlen = 0;
 	}
     }
-  else if (filename == pattern)
+  else if (filename == pattern
+	   || (filename == pattern + 1 && pattern[0] == '\\'
+	       && (flags & GLOB_NOESCAPE) == 0))
     {
-      /* "/pattern".  */
+      /* "/pattern" or "\\/pattern".  */
       dirname = "/";
       dirlen = 1;
       ++filename;
