---
 elf/dl-lookup.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/elf/dl-lookup.c
+++ b/elf/dl-lookup.c
@@ -423,7 +423,7 @@
 		     hash table.  */
 		  if (__builtin_expect (tab->size, 0))
 		    {
-		      assert (GLRO(dl_debug_mask) & DL_DEBUG_PRELINK);
+		      assert (GLRO_dl_debug_mask & DL_DEBUG_PRELINK);
 		      __rtld_lock_unlock_recursive (tab->lock);
 		      goto success;
 		    }
