Index: b/math/Makefile
===================================================================
--- a/math/Makefile	2008-08-08 19:08:16.000000000 +0200
+++ b/math/Makefile	2008-08-08 19:09:15.000000000 +0200
@@ -90,7 +90,7 @@
 # Rules for the test suite.
 tests = test-matherr test-fenv atest-exp atest-sincos atest-exp2 basic-test \
 	test-misc test-fpucw tst-definitions test-tgmath test-tgmath-ret \
-	bug-nextafter bug-nexttoward bug-tgmath1 test-tgmath-int test-tgmath2
+	bug-nextafter bug-nexttoward bug-tgmath1 test-tgmath-int
 # We do the `long double' tests only if this data type is available and
 # distinct from `double'.
 test-longdouble-yes = test-ldouble test-ildoubl
