--- glibc-2.5/include/stubs-prologue.h	2006-10-22 00:01:09.118211998 +0200
+++ glibc-2.5/include/stubs-prologue.h	2006-10-22 00:01:24.833196462 +0200
@@ -8,7 +8,7 @@
    every time called, usually setting errno to ENOSYS.  */
 
 #ifdef _LIBC
- #error Applications may not define the macro _LIBC
+# error Applications may not define the macro _LIBC
 #endif
 
 @ Placeholder line so we remember to keep the preceding blank line here.
