Index: glibc-2.6/nscd/nscd.conf
===================================================================
--- glibc-2.6.orig/nscd/nscd.conf
+++ glibc-2.6/nscd/nscd.conf
@@ -60,7 +60,9 @@
 	max-db-size		group		33554432
 	auto-propagate		group		yes
 
-	enable-cache		hosts		yes
+# hosts caching is broken with gethostby* calls, hence is now disabled
+# per default.  See /usr/share/doc/nscd/NEWS.Debian.
+	enable-cache		hosts		no
 	positive-time-to-live	hosts		3600
 	negative-time-to-live	hosts		20
 	suggested-size		hosts		211
