--- libc/sysdeps/unix/sysv/linux/sys/sysmacros.h
+++ libc/sysdeps/unix/sysv/linux/sys/sysmacros.h
@@ -62,9 +62,24 @@
 
 
 /* Access the functions with their traditional names.  */
-# define major(dev) gnu_dev_major (dev)
-# define minor(dev) gnu_dev_minor (dev)
-# define makedev(maj, min) gnu_dev_makedev (maj, min)
+static __inline unsigned int
+__NTH (major (unsigned long long int dev))
+{
+  return gnu_dev_major (dev);
+}
+
+static __inline unsigned int
+__NTH (minor (unsigned long long int dev))
+{
+  return gnu_dev_minor (dev);
+}
+
+static __inline unsigned long long int
+__NTH (makedev (unsigned int maj, unsigned int min))
+{
+  return gnu_dev_makedev (maj, min);
+}
 #endif
 
 #endif /* sys/sysmacros.h */
+
