---
 nptl/tst-tls6.sh            |    1 +-
 2 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/nptl/Makefile b/nptl/Makefile
index 71bf5ef..640a3e0 100644
--- a/nptl/Makefile
+++ b/nptl/Makefile
@@ -433,7 +433,7 @@ CFLAGS-tst-initializers1-c99.c = $(CFLAGS-tst-initializers1-<)
 CFLAGS-tst-initializers1-gnu89.c = $(CFLAGS-tst-initializers1-<)
 CFLAGS-tst-initializers1-gnu99.c = $(CFLAGS-tst-initializers1-<)
 
-tst-cancel7-ARGS = --command "$(built-program-cmd)"
+tst-cancel7-ARGS = --command "exec $(built-program-cmd)"
 tst-cancelx7-ARGS = $(tst-cancel7-ARGS)
 tst-umask1-ARGS = $(objpfx)tst-umask1.temp
 
@@ -544,7 +544,7 @@ $(objpfx)pt-initfini.s: pt-initfini.c
 		$(patsubst -f%,-fno-%,$(exceptions)) -o $@
 
 $(objpfx)tst-cleanup0.out: /dev/null $(objpfx)tst-cleanup0
-	$(make-test-out) 2>&1 | cmp - tst-cleanup0.expect >& $@
+	$(make-test-out) 2>&1 | cmp - tst-cleanup0.expect > $@ 2>&1
 
 # We only have one kind of startup code files.  Static binaries and
 # shared libraries are build using the PIC version.
