Index: elf/ldconfig.c
===================================================================
--- elf/ldconfig.c.orig
+++ elf/ldconfig.c
@@ -82,6 +82,7 @@
   int flag;
   ino64_t ino;
   dev_t dev;
+  time_t last_changed;
   struct dir_entry *next;
 };
 
@@ -101,6 +102,7 @@
 int opt_format = 1;
 
 /* Build cache.  */
+/* 0: don't build cache; 1: build if not up-to-date; 2: always build it. */
 static int opt_build_cache = 1;
 
 /* Generate links.  */
@@ -138,6 +140,7 @@
   { "print-cache", 'p', NULL, 0, N_("Print cache"), 0},
   { "verbose", 'v', NULL, 0, N_("Generate verbose messages"), 0},
   { NULL, 'N', NULL, 0, N_("Don't build cache"), 0},
+  { "force", 'F', NULL, 0, N_("Generate cache even if it looks up-to-date"), 0},
   { NULL, 'X', NULL, 0, N_("Don't generate links"), 0},
   { NULL, 'r', N_("ROOT"), 0, N_("Change to and use ROOT as root directory"), 0},
   { NULL, 'C', N_("CACHE"), 0, N_("Use CACHE as cache file"), 0},
@@ -148,6 +151,9 @@
   { NULL, 0, NULL, 0, NULL, 0 }
 };
 
+/* Most recent modification time of the configuration files */
+static time_t config_timestamp = 0;
+
 #define PROCINFO_CLASS static
 #include <dl-procinfo.c>
 
@@ -251,6 +257,9 @@
     case 'N':
       opt_build_cache = 0;
       break;
+    case 'F':
+      opt_build_cache = 2;
+      break;
     case 'n':
       opt_build_cache = 0;
       opt_only_cline = 1;
@@ -383,6 +392,7 @@
     {
       entry->ino = stat_buf.st_ino;
       entry->dev = stat_buf.st_dev;
+      entry->last_changed = stat_buf.st_mtime;
 
       add_single_dir (entry, 1);
     }
@@ -967,6 +977,26 @@
     }
 }
 
+/* Tells whether one of the directories is more recent than the cache */
+static bool
+cache_needs_update (void)
+{
+  struct stat64 stat_buf;
+  struct dir_entry *entry;
+
+  if (stat64 (cache_file, &stat_buf))
+    return true;
+  
+  if (config_timestamp >= stat_buf.st_mtime)
+    return true;
+  
+  for (entry = dir_entries; entry != NULL; entry = entry->next)
+    {
+      if (entry->last_changed >= stat_buf.st_mtime)
+        return true;
+    }
+  return false;
+}
 
 static void parse_conf_include (const char *config_file, unsigned int lineno,
 				bool do_chroot, const char *pattern);
@@ -1004,6 +1034,10 @@
       return;
     }
 
+  struct stat64 stat_buf;
+  if (!fstat64 (fileno(file), &stat_buf) && stat_buf.st_mtime > config_timestamp)
+    config_timestamp = stat_buf.st_mtime;
+
   /* No threads use this stream.  */
   __fsetlocking (file, FSETLOCKING_BYCALLER);
 
@@ -1297,6 +1331,9 @@
 	add_system_dir (LIBDIR);
     }
 
+  if (opt_build_cache == 1 && !cache_needs_update ())
+    return 0;
+
   search_dirs ();
 
   if (opt_build_cache)
