Index: b/sysdeps/unix/sysv/linux/check_pf.c
===================================================================
--- a/sysdeps/unix/sysv/linux/check_pf.c	2008-01-03 01:24:52.000000000 +0100
+++ b/sysdeps/unix/sysv/linux/check_pf.c	2008-08-08 18:38:06.000000000 +0200
@@ -25,6 +25,8 @@
 #include <string.h>
 #include <time.h>
 #include <unistd.h>
+#include <alloca.h>
+#include <stdlib.h>
 #include <sys/socket.h>
 
 #include <asm/types.h>
