--- x/posix/Makefile	2012-11-17 10:50:14.000000000 -0700
+++ x/posix/Makefile	2013-01-27 21:42:47.802737434 -0700
@@ -79,7 +79,7 @@
 tests		:= tstgetopt testfnm runtests \
 		   tst-preadwrite tst-preadwrite64 test-vfork regexbug1 \
 		   tst-getlogin tst-mmap tst-truncate \
-		   tst-truncate64 tst-fork tst-dir \
+		   tst-truncate64 tst-fork \
 		   tst-chmod bug-regex2 bug-regex3 bug-regex4 \
 		   tst-gnuglob bug-regex6 bug-regex7 \
 		   bug-regex8 bug-regex9 bug-regex10 bug-regex12 \
@@ -118,6 +118,9 @@
 tests-$(OPTION_EGLIBC_SPAWN) += tst-spawn
 tests-$(OPTION_EGLIBC_WORDEXP) += wordexp-test
 endif
+ifneq ($(shell stat --file-system --format=%t .),794c764f)
+tests		+= tst-dir
+endif
 others		:= getconf
 install-bin	:= getconf
 install-others-programs	:= $(inst_libexecdir)/getconf
