diff --git a/elf/dl-close.c b/elf/dl-close.c
index efb2b58..3991e65 100644
--- a/elf/dl-close.c
+++ b/elf/dl-close.c
@@ -231,7 +231,8 @@ _dl_close_worker (struct link_map *map)
     }
 
   /* Sort the entries.  */
-  _dl_sort_fini (ns->_ns_loaded, maps, nloaded, used, nsid);
+  if (nloaded > 1)
+    _dl_sort_fini (ns->_ns_loaded, maps, nloaded, used, nsid);
 
   /* Call all termination functions at once.  */
 #ifdef SHARED
diff --git a/elf/dl-deps.c b/elf/dl-deps.c
index d3c27f1..ed27090 100644
--- a/elf/dl-deps.c
+++ b/elf/dl-deps.c
@@ -623,11 +623,11 @@ Filters not supported with LD_TRACE_PRELINKING"));
 
   /* We can skip looking for the binary itself which is at the front
      of the search list.  */
-  assert (nlist > 1);
+  assert (nlist > 0);
   i = 1;
   bool seen[nlist];
   memset (seen, false, nlist * sizeof (seen[0]));
-  while (1)
+  while (nlist > 1)
     {
       /* Keep track of which object we looked at this round.  */
       seen[i] = true;

