Index: b/misc/mktemp.c
===================================================================
--- a/misc/mktemp.c	2008-08-08 19:08:17.000000000 +0200
+++ b/misc/mktemp.c	2008-08-08 19:09:13.000000000 +0200
@@ -33,4 +33,4 @@
   return template;
 }
 
-link_warning (mktemp, "the use of `mktemp' is dangerous, better use `mkstemp'")
+link_warning (mktemp, "the use of `mktemp' is dangerous, better use `mkstemp' or `mkdtemp'")
