--- glibc-2.3.6.orig/sysdeps/unix/sysv/linux/alpha/sys/user.h	2001-07-06 06:56:13.000000000 +0200
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/alpha/sys/user.h	2006-10-21 22:38:36.108043157 +0200
@@ -23,6 +23,7 @@
    only. Don't read too much into it. Don't use it for anything other
    than gdb/strace unless you know what you are doing. */
 
+#include <unistd.h>
 #include <asm/page.h>
 #include <asm/reg.h>
 
@@ -41,7 +42,7 @@
   char u_comm[32];				/* user command name */
 };
 
-#define NBPG			PAGE_SIZE
+#define NBPG			sysconf(_SC_PAGESIZE)
 #define UPAGES			1
 #define HOST_TEXT_START_ADDR	(u.start_code)
 #define HOST_DATA_START_ADDR	(u.start_data)
--- glibc-2.3.6.orig/sysdeps/unix/sysv/linux/ia64/sys/user.h	2002-08-25 05:57:38.000000000 +0200
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/ia64/sys/user.h	2006-10-21 22:38:50.554268504 +0200
@@ -21,6 +21,7 @@
 
 #include <features.h>
 #include <sys/types.h>
+#include <unistd.h>
 
 /* This definition comes directly from the kernel headers.  If
    anything changes in them this header has to be changed, too.  */
@@ -45,7 +46,7 @@
   char u_comm[32];				/* User command name.  */
 };
 
-#define NBPG			PAGE_SIZE
+#define NBPG			sysconf(_SC_PAGESIZE)
 #define UPAGES			1
 #define HOST_TEXT_START_ADDR	(u.start_code)
 #define HOST_DATA_START_ADDR	(u.start_data)
--- glibc-2.3.6.orig/ports/sysdeps/unix/sysv/linux/mips/sys/user.h	2004-11-24 05:37:35.000000000 +0100
+++ glibc-2.3.6/ports/sysdeps/unix/sysv/linux/mips/sys/user.h	2006-10-21 22:39:01.575151737 +0200
@@ -20,6 +20,7 @@
 #define _SYS_USER_H	1
 
 #include <sgidefs.h>
+#include <unistd.h>
 
 /* The whole purpose of this file is for GDB and GDB only.  Don't read
    too much into it.  Don't use it for anything other than GDB unless
@@ -207,8 +208,7 @@
 
 #endif
 
-#define PAGE_SHIFT		12
-#define PAGE_SIZE		(1UL << PAGE_SHIFT)
+#define PAGE_SIZE		sysconf(_SC_PAGESIZE)
 #define PAGE_MASK		(~(PAGE_SIZE-1))
 #define NBPG			PAGE_SIZE
 #define UPAGES			1
--- glibc-2.3.6.orig/sysdeps/unix/sysv/linux/x86_64/sys/user.h	2004-11-05 00:37:34.000000000 +0100
+++ glibc-2.3.6/sysdeps/unix/sysv/linux/x86_64/sys/user.h	2006-10-21 22:39:08.237872033 +0200
@@ -24,6 +24,7 @@
    you know what you are doing.  */
 
 #include <bits/wordsize.h>
+#include <unistd.h>
 
 #if __WORDSIZE == 64
 
@@ -164,8 +165,7 @@
 };
 #endif  /* __WORDSIZE */
 
-#define PAGE_SHIFT		12
-#define PAGE_SIZE		(1UL << PAGE_SHIFT)
+#define PAGE_SIZE		sysconf(_SC_PAGESIZE)
 #define PAGE_MASK		(~(PAGE_SIZE-1))
 #define NBPG			PAGE_SIZE
 #define UPAGES			1
