--- glibc-2.4/linuxthreads/Makefile.orig	2006-08-17 01:17:00.493091863 +0200
+++ glibc-2.4/linuxthreads/Makefile	2006-08-17 01:17:11.965907527 +0200
@@ -111,7 +111,7 @@
 	ex17 ex18 tst-cancel tst-context bug-sleep \
 	tst-cancel1 tst-cancel2 tst-cancel3 tst-cancel4 tst-cancel5 \
 	tst-cancel6 tst-cancel7 tst-cancel8 tst-popen tst-popen2 tst-attr1 \
-	tst-stack1 tst-align tst-tsd1 tst-sighandler1 tst-sighandler2
+	tst-stack1 tst-align tst-tsd1
 test-srcs = tst-signal
 # These tests are linked with libc before libpthread
 tests-reverse += tst-cancel5
