Index: b/iconvdata/run-iconv-test.sh
===================================================================
--- a/iconvdata/run-iconv-test.sh	2008-08-08 19:08:18.000000000 +0200
+++ b/iconvdata/run-iconv-test.sh	2008-08-08 19:09:03.000000000 +0200
@@ -130,7 +130,7 @@
     done
   fi
 
-  if test "$subset" == N; then
+  if test "$subset" = N; then
     echo $ac_n "      suntzu: ASCII -> $to -> ASCII $ac_c"
     $PROG -f ASCII -t $to testdata/suntzus |
     $PROG -f $to -t ASCII > $temp1 ||
Index: b/nptl/tst-tls6.sh
===================================================================
--- a/nptl/tst-tls6.sh	2008-08-08 19:08:18.000000000 +0200
+++ b/nptl/tst-tls6.sh	2008-08-08 19:09:03.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 common_objpfx=$1; shift
 elf_objpfx=$1; shift
