Index: b/sysdeps/unix/opendir.c
===================================================================
--- a/sysdeps/unix/opendir.c	2008-08-08 19:08:17.000000000 +0200
+++ b/sysdeps/unix/opendir.c	2008-08-08 19:09:14.000000000 +0200
@@ -146,6 +146,7 @@
 #ifdef __ASSUME_O_CLOEXEC
 # define check_have_o_cloexec(fd) 1
 #else
+#ifdef O_CLOEXEC
 static int
 check_have_o_cloexec (int fd)
 {
@@ -154,6 +155,7 @@
   return __have_o_cloexec > 0;
 }
 #endif
+#endif
 
 
 DIR *
