Index: iconvdata/run-iconv-test.sh
===================================================================
--- iconvdata/run-iconv-test.sh.orig
+++ iconvdata/run-iconv-test.sh
@@ -130,7 +130,7 @@
     done
   fi
 
-  if test "$subset" == N; then
+  if test "$subset" = N; then
     echo $ac_n "      suntzu: ASCII -> $to -> ASCII $ac_c"
     $PROG -f ASCII -t $to testdata/suntzus |
     $PROG -f $to -t ASCII > $temp1 ||
Index: linuxthreads/tst-tls2.sh
===================================================================
--- linuxthreads/tst-tls2.sh.orig
+++ linuxthreads/tst-tls2.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 common_objpfx=$1; shift
 elf_objpfx=$1; shift
Index: nptl/tst-tls6.sh
===================================================================
--- nptl/tst-tls6.sh.orig
+++ nptl/tst-tls6.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 common_objpfx=$1; shift
 elf_objpfx=$1; shift
