Index: b/nscd/nscd.conf
===================================================================
--- a/nscd/nscd.conf	2007-11-06 01:50:48.000000000 +0100
+++ b/nscd/nscd.conf	2008-08-08 18:38:17.000000000 +0200
@@ -60,7 +60,9 @@
 	max-db-size		group		33554432
 	auto-propagate		group		yes
 
-	enable-cache		hosts		yes
+# hosts caching is broken with gethostby* calls, hence is now disabled
+# per default.  See /usr/share/doc/nscd/NEWS.Debian.
+	enable-cache		hosts		no
 	positive-time-to-live	hosts		3600
 	negative-time-to-live	hosts		20
 	suggested-size		hosts		211
