Index: sysdeps/generic/framestate.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/generic/framestate.c,v
retrieving revision 1.4
diff -u -p -r1.4 framestate.c
--- sysdeps/generic/framestate.c	21 Feb 2004 09:07:05 -0000	1.4
+++ sysdeps/generic/framestate.c	28 Jul 2007 21:19:14 -0000
@@ -25,6 +25,10 @@
 #include <unwind-dw2.c>
 #undef __frame_state_for
 
+#ifndef LIBGCC_S_SO
+#define LIBGCC_S_SO "libgcc_s.so.1"
+#endif
+
 typedef struct frame_state * (*framesf)(void *pc, struct frame_state *);
 struct frame_state *__frame_state_for (void *pc,
 				       struct frame_state *frame_state);
@@ -36,7 +40,7 @@ __frame_state_for (void *pc, struct fram
 
   if (frame_state_for == NULL)
     {
-      void *handle = __libc_dlopen ("libgcc_s.so.1");
+      void *handle = __libc_dlopen (LIBGCC_S_SO);
 
       if (handle == NULL
 	  || (frame_state_for
