Index: configure
===================================================================
RCS file: /cvs/glibc/libc/configure,v
retrieving revision 1.462
diff -u -d -p -r1.462 configure
--- configure	24 Jan 2008 20:21:23 -0000	1.462
+++ configure	31 Jan 2008 02:24:00 -0000
@@ -5066,7 +5066,7 @@ echo "$as_me: WARNING:
 # thing on a system that doesn't need fixincludes.  (Not presently a problem.)
 if test -n "$sysheaders"; then
   ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  SYSINCLUDES="-nostdinc -isystem $ccheaders -isystem $ccheaders-fixed \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     cxxversion=`$CXX -dumpversion 2>&5` &&
Index: configure.in
===================================================================
RCS file: /cvs/glibc/libc/configure.in,v
retrieving revision 1.472
diff -u -d -p -r1.472 configure.in
--- configure.in	24 Jan 2008 20:21:23 -0000	1.472
+++ configure.in	31 Jan 2008 02:24:00 -0000
@@ -915,7 +915,7 @@ test -n "$aux_missing" && AC_MSG_WARN([
 # thing on a system that doesn't need fixincludes.  (Not presently a problem.)
 if test -n "$sysheaders"; then
   ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  SYSINCLUDES="-nostdinc -isystem $ccheaders -isystem $ccheaders-fixed \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     cxxversion=`$CXX -dumpversion 2>&AS_MESSAGE_LOG_FD` &&
