--- iconv/gconv_charset.h	2005-12-21 08:27:27.000000000 +0100
+++ iconv/gconv_charset.h	2007-04-22 00:57:25.000000000 +0200
@@ -30,7 +30,7 @@
   while (*s != '\0')
     {
       if (__isalnum_l (*s, _nl_C_locobj_ptr)
-	  || *s == '_' || *s == '-' || *s == '.' || *s == ',')
+	  || *s == '_' || *s == '-' || *s == '.' || *s == ',' || *s == ':')
 	*wp++ = __toupper_l (*s, _nl_C_locobj_ptr);
       else if (*s == '/')
 	{
--- iconv/iconv_open.c	2001-11-29 05:57:52.000000000 +0100
+++ iconv/iconv_open.c	2007-04-22 00:59:08.000000000 +0200
@@ -38,7 +38,7 @@
   int res;
 
   /* Normalize the name.  We remove all characters beside alpha-numeric,
-     '_', '-', '/', and '.'.  */
+     '_', '-', '/', '.' and ':'.  */
   tocode_len = strlen (tocode);
   tocode_conv = (char *) alloca (tocode_len + 3);
   strip (tocode_conv, tocode);
