http://sourceware.org/bugzilla/show_bug.cgi?id=4026

Index: sysdeps/unix/sysv/linux/clock_settime.c
===================================================================
--- sysdeps/unix/sysv/linux/clock_settime.c.orig
+++ sysdeps/unix/sysv/linux/clock_settime.c
@@ -18,6 +18,7 @@
 
 #include <errno.h>
 #include <sysdep.h>
+#include <time.h>
 
 #include "kernel-posix-cpu-timers.h"
 #include <kernel-features.h>
