Index: glibc-2.5/stdio-common/vfprintf.c
===================================================================
--- glibc-2.5.orig/stdio-common/vfprintf.c
+++ glibc-2.5/stdio-common/vfprintf.c
@@ -1161,12 +1161,11 @@
 		    /* In case we have a multibyte character set the	      \
 		       situation is more compilcated.  We must not copy	      \
 		       bytes at the end which form an incomplete character. */\
-		    wchar_t ignore[prec];				      \
 		    const char *str2 = string;				      \
 		    mbstate_t ps;					      \
 									      \
 		    memset (&ps, '\0', sizeof (ps));			      \
-		    if (__mbsnrtowcs (ignore, &str2, prec, prec, &ps)	      \
+		    if (__mbsnrtowcs (NULL, &str2, prec, prec, &ps)	      \
 			== (size_t) -1)					      \
 		      {							      \
 			done = -1;					      \
