diff --git a/elf/Makefile b/elf/Makefile
index 84f8397..6d4b365 100644
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -849,6 +849,9 @@ LDFLAGS-tst-execstack-needed = -Wl,-z,noexecstack
 LDFLAGS-tst-execstack-prog = -Wl,-z,execstack
 endif
 
+LDFLAGS-tst-array2 = $(no-as-needed)
+LDFLAGS-tst-array5 = $(no-as-needed)
+
 $(objpfx)tst-array1.out: tst-array1.exp $(objpfx)tst-array1
 	$(built-program-cmd) > $@
 	cmp $@ tst-array1.exp > /dev/null
