--- libc/sysdeps/unix/sysv/linux/getcwd.c.orig	2006-03-10 17:42:19.000000000 +0100
+++ libc/sysdeps/unix/sysv/linux/getcwd.c	2006-03-10 17:45:39.000000000 +0100
@@ -127,7 +127,7 @@
       /* It should never happen that the `getcwd' syscall failed because
 	 the buffer is too small if we allocated the buffer ourselves
 	 large enough.  */
-      assert (errno != ERANGE || buf != NULL || size != 0);
+/*      assert (errno != ERANGE || buf != NULL || size != 0); */
 
 #  ifndef NO_ALLOCATION
       if (buf == NULL)
