Index: sysdeps/mach/hurd/getresuid.c
===================================================================
--- sysdeps/mach/hurd/getresuid.c	2002-10-13 21:03:11.000000000 -0400
+++ sysdeps/mach/hurd/getresuid.c	2006-02-12 11:08:46.000000000 -0500
@@ -25,7 +25,7 @@
 /* Fetch the effective user ID, real user ID, and saved-set user ID,
    of the calling process.  */
 int
-__getresuid (uid_t *euid, uid_t *ruid, uid_t *suid)
+__getresuid (uid_t *ruid, uid_t *euid, uid_t *suid)
 {
   error_t err;
   uid_t eff, real, saved;
