Index: glibc/sysdeps/i386/dl-tlsdesc.S
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/i386/dl-tlsdesc.S,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -p -r1.1 -r1.2
--- glibc/sysdeps/i386/dl-tlsdesc.S	13 May 2008 05:33:06 -0000	1.1
+++ glibc/sysdeps/i386/dl-tlsdesc.S	6 Dec 2008 00:15:17 -0000	1.2
@@ -128,8 +128,7 @@ _dl_tlsdesc_dynamic:
 .Lslow:
 	cfi_adjust_cfa_offset (28)
 	movl	%ebx, 16(%esp)
-	call	__i686.get_pc_thunk.bx
-	addl	$_GLOBAL_OFFSET_TABLE_, %ebx
+	LOAD_PIC_REG (bx)
 	call	___tls_get_addr@PLT
 	movl	16(%esp), %ebx
 	jmp	.Lret
