Index: glibc-2.23.90+20160507.87523e9/libpthread/Versions
===================================================================
--- glibc-2.23.90+20160507.87523e9.orig/libpthread/Versions
+++ glibc-2.23.90+20160507.87523e9/libpthread/Versions
@@ -35,10 +35,12 @@ libc {
     pthread_self; pthread_setcancelstate; pthread_setcanceltype;
     __pthread_get_cleanup_stack;
   }
+  GLIBC_2.24 {
+    __register_atfork;
+  }
   GLIBC_PRIVATE {
     __libc_alloca_cutoff;
     __libc_pthread_init;
-    __register_atfork;
   }
 }
 
