Index: b/sysdeps/mach/hurd/bits/fcntl.h
===================================================================
--- a/sysdeps/mach/hurd/bits/fcntl.h	2007-11-18 23:51:16.000000000 +0100
+++ b/sysdeps/mach/hurd/bits/fcntl.h	2008-08-08 18:37:59.000000000 +0200
@@ -22,6 +22,8 @@
 # error "Never use <bits/fcntl.h> directly; include <fcntl.h> instead."
 #endif
 
+#include <sys/types.h>
+
 /* File access modes.  These are understood by io servers; they can be
    passed in `dir_lookup', and are returned by `io_get_openmodes'.
    Consequently they can be passed to `open', `hurd_file_name_lookup', and
