Index: glibc/bits/resource.h
===================================================================
RCS file: /cvs/glibc/libc/bits/resource.h,v
retrieving revision 1.8
retrieving revision 1.9
diff -u -p -r1.8 -r1.9
--- glibc/bits/resource.h	5 Mar 2006 22:48:10 -0000	1.8
+++ glibc/bits/resource.h	3 Jan 2009 21:04:01 -0000	1.9
@@ -185,6 +185,9 @@ struct rusage
 enum __priority_which
   {
     PRIO_PROCESS = 0,           /* WHO is a process ID.  */
+#define PRIO_PROCESS PRIO_PROCESS
     PRIO_PGRP = 1,              /* WHO is a process group ID.  */
+#define PRIO_PGRP PRIO_PGRP
     PRIO_USER = 2               /* WHO is a user ID.  */
+#define PRIO_USER PRIO_USER
   };
