commit cd9fa98583b51812e906f1e85cc6e370bb4aff87
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Thu May 10 13:10:21 2012 -0700

    Hurd: Hurd: dl-sysdep.c: Include code only #ifdef SHARED.

diff --git a/sysdeps/mach/hurd/dl-sysdep.c b/sysdeps/mach/hurd/dl-sysdep.c
index 26a212e..13b0010 100644
--- a/sysdeps/mach/hurd/dl-sysdep.c
+++ b/sysdeps/mach/hurd/dl-sysdep.c
@@ -16,6 +16,10 @@
    License along with the GNU C Library; if not, see
    <http://www.gnu.org/licenses/>.  */
 
+/* In the static library, this is all handled by dl-support.c
+   or by the vanilla definitions in the rest of the C library.  */
+#ifdef SHARED
+
 #include <hurd.h>
 #include <link.h>
 #include <unistd.h>
@@ -667,3 +671,5 @@ _dl_init_first (int argc, ...)
 {
   /* This no-op definition only gets used if libc is not linked in.  */
 }
+
+#endif /* SHARED */
