Index: sysdeps/mach/hurd/bits/fcntl.h
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/bits/fcntl.h,v
retrieving revision 1.13
retrieving revision 1.14
diff -u -p -r1.13 -r1.14
--- a/sysdeps/mach/hurd/bits/fcntl.h	17 Oct 2007 22:32:35 -0000	1.13
+++ b/sysdeps/mach/hurd/bits/fcntl.h	18 Nov 2007 22:51:16 -0000	1.14
@@ -71,7 +71,6 @@
 #define O_SYNC		O_FSYNC
 #ifdef __USE_GNU
 # define O_NOATIME	0x0800	/* Don't set access time on read (owner).  */
-# define O_CLOEXEC	0x00010000 /* Set FD_CLOEXEC.  */
 #endif
 #ifdef	__USE_MISC
 # define O_SHLOCK	0x00020000 /* Open with shared file lock.  */
@@ -118,6 +117,9 @@
    once the file has been opened.  */
 
 #define	O_TRUNC		0x00010000 /* Truncate file to zero length.  */
+#ifdef __USE_GNU
+# define O_CLOEXEC	0x00400000 /* Set FD_CLOEXEC.  */
+#endif
 
 
 /* Controlling terminal flags.  These are understood only by `open',
