Index: glibc/sysdeps/mach/hurd/bits/fcntl.h
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/bits/fcntl.h,v
retrieving revision 1.14
diff -u -p -r1.14 fcntl.h
--- glibc/sysdeps/mach/hurd/bits/fcntl.h	18 Nov 2007 22:51:16 -0000	1.14
+++ glibc/sysdeps/mach/hurd/bits/fcntl.h	14 Jul 2008 00:47:13 -0000
@@ -22,6 +22,8 @@
 # error "Never use <bits/fcntl.h> directly; include <fcntl.h> instead."
 #endif
 
+#include <sys/types.h>
+
 /* File access modes.  These are understood by io servers; they can be
    passed in `dir_lookup', and are returned by `io_get_openmodes'.
    Consequently they can be passed to `open', `hurd_file_name_lookup', and
