Index: glibc/libpthread/Makefile
===================================================================
--- glibc.orig/libpthread/Makefile
+++ glibc/libpthread/Makefile
@@ -235,9 +235,6 @@ CPPFLAGS += \
 ifeq ($(IN_GLIBC),no)
 CPPFLAGS += \
 	  -imacros $(srcdir)/include/libc-symbols.h
-else
-CPPFLAGS += \
-	  -imacros libc-symbols.h
 endif
 
 ifeq ($(IN_GLIBC),yes)
@@ -291,7 +288,7 @@ include ../Rules
 # a statically-linked program that hasn't already loaded it.
 # Depend on ld.so too to get proper versions of ld.so symbols.
 $(objpfx)libpthread.so: $(libc-link.so) $(common-objpfx)libc_nonshared.a \
-			$(if $(filter yes,$(elf)), $(elfobjdir)/ld.so) \
+			$(if $(filter yes,$(elf)), $(elf-objpfx)/ld.so) \
 			$(common-objpfx)/mach/libmachuser.so \
 			$(common-objpfx)/hurd/libhurduser.so
 endif
