Index: glibc/hurd/report-wait.c
===================================================================
RCS file: /cvs/glibc/libc/hurd/report-wait.c,v
retrieving revision 1.15
retrieving revision 1.16
diff -u -p -r1.15 -r1.16
--- glibc/hurd/report-wait.c	29 Dec 2005 10:38:12 -0000	1.15
+++ glibc/hurd/report-wait.c	7 Jan 2009 01:59:26 -0000	1.16
@@ -1,5 +1,5 @@
 /* Report on what a thread in our task is waiting for.
-   Copyright (C) 1996,1997,1999,2002,2005 Free Software Foundation, Inc.
+   Copyright (C) 1996,1997,1999,2002,2005,2009 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -30,7 +30,7 @@ static char *
 describe_number (string_t description, const char *flavor, long int i)
 {
   unsigned long int j;
-  char *p = flavor ? description : __stpcpy (description, flavor);
+  char *p = flavor == NULL ? description : __stpcpy (description, flavor);
   char *end;
 
   /* Handle sign.  */
