Index: glibc/hurd/hurdsig.c
===================================================================
RCS file: /cvs/glibc/libc/hurd/hurdsig.c,v
retrieving revision 1.160
retrieving revision 1.161
diff -u -p -r1.160 -r1.161
--- glibc/hurd/hurdsig.c	8 Mar 2008 21:47:23 -0000	1.160
+++ glibc/hurd/hurdsig.c	7 Jan 2009 01:59:35 -0000	1.161
@@ -1134,6 +1134,7 @@ signal_allowed (int signo, mach_port_t r
 	      }
 	    _hurd_port_free (&_hurd_dtable[d]->port, &ulink, port);
 	  }
+	__mutex_unlock (&_hurd_dtable_lock);
 	/* If we found a lucky winner, we've set D to -1 in the loop.  */
 	if (lucky)
 	  goto win;
