diff --git a/ChangeLog b/ChangeLog
index add5128cb7..b8184b658d 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2018-02-04  Samuel Thibault  <samuel.thibault@ens-lyon.org>
+
+	* stdlib/test-atexit-race-common.c [!defined PTHREAD_STACK_MIN]: Do
+	not check against PTHREAD_STACK_MIN.
+
 2018-02-02  Sean McKean  <smckean83@gmail.com>
 
 	[BZ #22735]
diff --git a/stdlib/test-atexit-race-common.c b/stdlib/test-atexit-race-common.c
index 8adf1330bf..63bde390b4 100644
--- a/stdlib/test-atexit-race-common.c
+++ b/stdlib/test-atexit-race-common.c
@@ -39,7 +39,10 @@
 const size_t kNumThreads = 1024;
 const size_t kNumHandlers = 1024;
 const size_t kStacksize =
-	0x20000 < PTHREAD_STACK_MIN ? PTHREAD_STACK_MIN : 0x20000;
+#ifdef PTHREAD_STACK_MIN
+	0x20000 < PTHREAD_STACK_MIN ? PTHREAD_STACK_MIN :
+#endif
+		0x20000;
 
 static void *
 threadfunc (void *unused)
