commit b794e16bcc9426e9e05ec465c6c8bab2902a50ed
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sun Sep 3 01:08:49 2017 +0200

    sysdeps/generic/sigsetops.h: Remove spurious #endif

diff --git a/sysdeps/generic/sigsetops.h b/sysdeps/generic/sigsetops.h
index e8e5553981..ea2c5b5767 100644
--- a/sysdeps/generic/sigsetops.h
+++ b/sysdeps/generic/sigsetops.h
@@ -50,7 +50,6 @@
     *(dest) = *(left) | *(right);		\
     (void)0;					\
   }))
-#endif
 
 /* These macros needn't check for a bogus signal number;
    checking is done in the non-__ versions.  */
