diff --git a/resolv/tst-resolv-res_init-skeleton.c b/resolv/tst-resolv-res_init-skeleton.c
index 3b7b4129e3..a1bf110910 100644
--- a/resolv/tst-resolv-res_init-skeleton.c
+++ b/resolv/tst-resolv-res_init-skeleton.c
@@ -341,11 +341,15 @@ run_res_init (void *closure)
     setenv ("RES_OPTIONS", ctx->t->res_options, 1);
   if (ctx->t->hostname != NULL)
     {
+#ifdef CLONE_NEWUTS
       /* This test needs its own namespace, to avoid changing the host
          name for the parent, too.  */
       TEST_VERIFY_EXIT (unshare (CLONE_NEWUTS) == 0);
       if (sethostname (ctx->t->hostname, strlen (ctx->t->hostname)) != 0)
         FAIL_EXIT1 ("sethostname (\"%s\"): %m", ctx->t->hostname);
+#else
+      FAIL_UNSUPPORTED ("clone (CLONE_NEWUTS) not supported");
+#endif
     }
 
   switch (ctx->init)
