Actually part of commit 988f991b5069f09d793c887618e84a196b100416
('Remove non cancellable sigsuspend definition')

Index: glibc-2.26/sysdeps/mach/hurd/sigsuspend.c
===================================================================
--- glibc-2.26.orig/sysdeps/mach/hurd/sigsuspend.c
+++ glibc-2.26/sysdeps/mach/hurd/sigsuspend.c
@@ -79,5 +79,4 @@ __sigsuspend (const sigset_t *set)
   return -1;
 }
 libc_hidden_def (__sigsuspend)
-strong_alias (__sigsuspend, sigsuspend_not_cancel)
 weak_alias (__sigsuspend, sigsuspend)
