Index: glibc-2.27/sysdeps/pthread/timer_routines.c
===================================================================
--- glibc-2.27.orig/sysdeps/pthread/timer_routines.c
+++ glibc-2.27/sysdeps/pthread/timer_routines.c
@@ -463,10 +463,14 @@ int
 __timer_thread_start (struct thread_node *thread)
 {
   int retval = 1;
+  sigset_t set, oset;
 
   assert (!thread->exists);
   thread->exists = 1;
 
+  sigfillset (&set);
+  pthread_sigmask (SIG_SETMASK, &set, &oset);
+
   if (pthread_create (&thread->id, &thread->attr,
 		      (void *(*) (void *)) thread_func, thread) != 0)
     {
@@ -474,6 +478,8 @@ __timer_thread_start (struct thread_node
       retval = -1;
     }
 
+  pthread_sigmask (SIG_SETMASK, &oset, NULL);
+
   return retval;
 }
 
