Index: glibc-2.27/sysdeps/mach/hurd/Implies
===================================================================
--- glibc-2.27.orig/sysdeps/mach/hurd/Implies
+++ glibc-2.27/sysdeps/mach/hurd/Implies
@@ -3,3 +3,9 @@
 gnu
 # The Hurd provides a rough superset of the functionality of 4.4 BSD.
 unix/bsd
+libpthread
+../libpthread/sysdeps/mach/hurd
+../libpthread/sysdeps/hurd
+../libpthread/sysdeps/mach
+../libpthread/sysdeps/posix
+../libpthread/sysdeps/generic
Index: glibc-2.27/sysdeps/mach/hurd/i386/Implies
===================================================================
--- glibc-2.27.orig/sysdeps/mach/hurd/i386/Implies
+++ glibc-2.27/sysdeps/mach/hurd/i386/Implies
@@ -1 +1,3 @@
 mach/hurd/x86
+../libpthread/sysdeps/mach/hurd/i386
+../libpthread/sysdeps/i386
Index: glibc-2.27/sysdeps/mach/hurd/Subdirs
===================================================================
--- glibc-2.27.orig/sysdeps/mach/hurd/Subdirs
+++ glibc-2.27/sysdeps/mach/hurd/Subdirs
@@ -7,3 +7,5 @@
 # way with more somewhat expensive `make' invocations.
 
 first hurd
+
+libpthread
Index: glibc-2.27/libpthread/sysdeps/pthread/pthread-functions.h
===================================================================
--- glibc-2.27.orig/libpthread/sysdeps/pthread/pthread-functions.h
+++ glibc-2.27/libpthread/sysdeps/pthread/pthread-functions.h
@@ -133,7 +133,7 @@ struct pthread_functions
 extern struct pthread_functions __libc_pthread_functions attribute_hidden;
 extern int __libc_pthread_functions_init attribute_hidden;
 
-void __libc_pthread_init (const struct pthread_functions *functions) internal_function;
+void __libc_pthread_init (const struct pthread_functions *functions);
 
 # define PTHFCT_CALL(fct, params) \
     __libc_pthread_functions.fct params
Index: glibc-2.27/libpthread/libc_pthread_init.c
===================================================================
--- glibc-2.27.orig/libpthread/libc_pthread_init.c
+++ glibc-2.27/libpthread/libc_pthread_init.c
@@ -21,7 +21,6 @@
 #include <pthread-functions.h>
 
 void
-internal_function
 __libc_pthread_init (const struct pthread_functions *functions)
 {
 #ifdef SHARED
Index: glibc-2.27/sysdeps/libpthread/Makeconfig
===================================================================
--- /dev/null
+++ glibc-2.27/sysdeps/libpthread/Makeconfig
@@ -0,0 +1 @@
+include $(..)libpthread/Makeconfig
