This dates when libpthread was moved to glibc.


--- a/Versions.def.orig	2012-04-25 15:45:24.000000000 +0000
+++ b/Versions.def	2012-04-25 15:45:31.000000000 +0000
@@ -38,6 +38,7 @@
 %endif
   GLIBC_PRIVATE
   GLIBC_2.13_DEBIAN_19
+  GLIBC_2.13_DEBIAN_31
 }
 libcrypt {
   GLIBC_2.0
--- a/libpthread/Versions.orig	2012-04-25 15:43:25.000000000 +0000
+++ b/libpthread/Versions	2012-04-25 15:43:40.000000000 +0000
@@ -1,5 +1,5 @@
 libc {
-  GLIBC_2.13 {
+  GLIBC_2.13_DEBIAN_31 {
     pthread_attr_destroy; pthread_attr_getdetachstate;
     pthread_attr_getinheritsched; pthread_attr_getschedparam;
     pthread_attr_getschedpolicy; pthread_attr_getscope; pthread_attr_init;
