Index: glibc-2.31/sysdeps/mach/hurd/bits/posix_opt.h
===================================================================
--- glibc-2.31.orig/sysdeps/mach/hurd/bits/posix_opt.h
+++ glibc-2.31/sysdeps/mach/hurd/bits/posix_opt.h
@@ -126,8 +126,8 @@
 /* POSIX.4 shared memory objects are supported (using regular files).  */
 #define _POSIX_SHARED_MEMORY_OBJECTS	_POSIX_MAPPED_FILES
 
-/* CPU-time clocks support needs to be checked at runtime.  */
-#define _POSIX_CPUTIME	0
+/* CPU-time clocks are not yet supported.  */
+#define _POSIX_CPUTIME	-1
 
 /* Clock support in threads must be also checked at runtime.  */
 #define _POSIX_THREAD_CPUTIME	0
