For now errno is still a threadvar variable.

Index: b/include/errno.h
===================================================================
--- a/include/errno.h	2006-10-28 01:11:42.000000000 +0200
+++ b/include/errno.h	2008-08-08 18:37:54.000000000 +0200
@@ -21,7 +21,7 @@
 
 #  include <tls.h>
 
-#  if USE___THREAD
+#  if USE___THREAD && !defined(__GNU__)
 #   undef  errno
 #   ifndef NOT_IN_libc
 #    define errno __libc_errno
