--- glibc/sysdeps/i386/tlsdesc.sym.orig	2009-01-27 22:50:04.000000000 +0000
+++ glibc/sysdeps/i386/tlsdesc.sym	2009-01-27 22:50:27.000000000 +0000
@@ -8,7 +8,11 @@
 
 -- Abuse tls.h macros to derive offsets relative to the thread register.
 
+#if 0
 DTV_OFFSET			offsetof(struct pthread, header.dtv)
+#else
+DTV_OFFSET			offsetof(tcbhead_t, dtv)
+#endif
 
 TLSDESC_ARG			offsetof(struct tlsdesc, arg)
 
