Index: glibc/sysdeps/mach/hurd/Makefile
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/Makefile,v
retrieving revision 1.95
diff -u -p -r1.95 Makefile
--- glibc/sysdeps/mach/hurd/Makefile	17 Oct 2004 20:00:25 -0000	1.95
+++ glibc/sysdeps/mach/hurd/Makefile	6 Jan 2009 15:15:13 -0000
@@ -206,4 +206,9 @@ ifeq ($(subdir),sunrpc)
 sysdep_headers += nfs/nfs.h
 endif
 
+ifeq ($(subdir),socket)
+sysdep_headers += net/ethernet.h net/if_arp.h net/if_ether.h net/if_ppp.h \
+		  net/route.h
+endif
+
 endif	# in-Makerules
