Index: include/stubs-biarch.h
===================================================================
RCS file: /cvs/glibc/libc/include/stubs-biarch.h,v
retrieving revision 1.1
diff -u -p -r1.1 stubs-biarch.h
--- glibc/include/stubs-biarch.h	11 Feb 2005 10:17:01 -0000	1.1
+++ glibc/include/stubs-biarch.h	26 May 2008 20:51:22 -0000
@@ -10,3 +10,5 @@
 #else
 # error "unexpected value for __WORDSIZE macro"
 #endif
+
+#include <gnu/stubs-pthread.h>
