--- argp/argp-help.c.orig
+++ argp/argp-help.c
@@ -42,6 +42,7 @@
 # endif
 #endif
 
+#include <stdbool.h>
 #include <stddef.h>
 #include <stdlib.h>
 #include <string.h>
--- posix/wordexp.c.orig
+++ posix/wordexp.c
@@ -28,6 +28,7 @@
 #include <paths.h>
 #include <pwd.h>
 #include <signal.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
--- stdlib/fmtmsg.c.orig
+++ stdlib/fmtmsg.c
@@ -19,6 +19,7 @@
 
 #include <fmtmsg.h>
 #include <bits/libc-lock.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
