---
 hurd/report-wait.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/hurd/report-wait.c
+++ b/hurd/report-wait.c
@@ -30,7 +30,7 @@
 describe_number (string_t description, const char *flavor, long int i)
 {
   unsigned long int j;
-  char *p = flavor ? description : __stpcpy (description, flavor);
+  char *p = !flavor ? description : __stpcpy (description, flavor);
   char *end;
 
   /* Handle sign.  */
