diff --git a/sysdeps/mach/Makefile b/sysdeps/mach/Makefile
index b47cdc6..3323e81 100644
--- a/sysdeps/mach/Makefile
+++ b/sysdeps/mach/Makefile
@@ -35,7 +35,7 @@ endif
 # because it's different in Darwin and the conditional crap is
 # too much trouble.  This should suffice for getting the mach/Makefile
 # rules invoked when they need to be.
-mach-before-compile := $(mach-objpfx)mach-shortcuts.h \
+mach-before-compile := $(mach-objpfx)mach-shortcuts.h $(mach-objpfx)mach/mach_interface.h \
 		       $(patsubst %,$(mach-objpfx)mach/mach_%.h,\
 				  port host)
 
@@ -43,7 +43,7 @@ ifneq (mach,$(subdir))
 # This patsubst generates patterns like `m%h-shortcuts.h', which are damn
 # likely to match just the corresponding particular file we want.
 $(patsubst mach%,m\%h%,$(mach-before-compile)): # Run only if doesn't exist.
-	$(MAKE) -C $(..)mach mach-before-compile no_deps=t generating=t
+	$(MAKE) -C $(..)mach subdir=mach mach-before-compile no_deps=t generating=t
 
 before-compile += $(mach-before-compile)
 endif
diff --git a/sysdeps/mach/hurd/Makefile b/sysdeps/mach/hurd/Makefile
index b528815..244ac4b 100644
--- a/sysdeps/mach/hurd/Makefile
+++ b/sysdeps/mach/hurd/Makefile
@@ -48,13 +48,13 @@ hurd-objpfx = $(common-objpfx)hurd/
 before-compile += $(patsubst %,$(hurd-objpfx)hurd/%.h,auth io fs process)
 $(patsubst %,$(hurd-objpfx)hurd/%.%,auth io fs process): \
   $(common-objpfx)mach/mach-shortcuts.h
-	$(MAKE) -C $(..)hurd before-compile no_deps=t
+	$(MAKE) -C $(..)hurd subdir=hurd before-compile no_deps=t
 endif
 
 # Hurd profil.c includes this file, so give a rule to make it.
 ifeq ($(subdir),gmon)
 $(common-objpfx)hurd/../mach/RPC_task_get_sampled_pcs.c:
-	$(MAKE) -C $(..)mach before-compile no_deps=t
+	$(MAKE) -C $(..)mach subdir=mach before-compile no_deps=t
 endif
 
 
