diff --git a/sysdeps/mach/hurd/cthreads.c b/sysdeps/mach/hurd/cthreads.c
index 5b5ed51..690d132 100644
--- a/sysdeps/mach/hurd/cthreads.c
+++ b/sysdeps/mach/hurd/cthreads.c
@@ -56,6 +56,7 @@ cthread_setspecific (key, val)
 /* Call cthread_getspecific which gets a pointer to the return value instead
    of just returning it.  */
 void *
+weak_function
 __libc_getspecific (key)
      cthread_key_t key;
 {
