From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Subject: [PATCH] We do have pthread types.

---
 sysdeps/mach/hurd/bits/pthreadtypes.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/sysdeps/mach/hurd/bits/pthreadtypes.h b/sysdeps/mach/hurd/bits/pthreadtypes.h
new file mode 100644
index 0000000..741f3eb
--- /dev/null
+++ b/sysdeps/mach/hurd/bits/pthreadtypes.h
@@ -0,0 +1 @@
+#include <pthread/pthreadtypes.h>
-- 
tg: (0234227..) t/pthread_types (depends on: baseline)
