--- a/bits/mman.h.orig	2012-06-22 21:16:15.919833847 +0200
+++ b/bits/mman.h	2012-06-22 21:16:23.323681644 +0200
@@ -86,10 +86,12 @@
 #define MS_INVALIDATE	2		/* Invalidate the caches.  */
 
 /* Flags for `mremap'.  */
+#if 0
 #ifdef __USE_GNU
 # define MREMAP_MAYMOVE	1		/* Mapping address may change.  */
 # define MREMAP_FIXED	2		/* Fifth argument sets new address.  */
 #endif
+#endif
 
 /* Flags for `mlockall' (can be OR'd together).  */
 #define MCL_CURRENT	1		/* Lock all currently mapped pages.  */
