Index: glibc/libpthread/sysdeps/i386/bits/semaphore.h
===================================================================
--- /dev/null
+++ glibc/libpthread/sysdeps/i386/bits/semaphore.h
@@ -0,0 +1 @@
+#include "../../generic/bits/semaphore.h"
Index: glibc/libpthread/sysdeps/pthread/semaphore.h
===================================================================
--- /dev/null
+++ glibc/libpthread/sysdeps/pthread/semaphore.h
@@ -0,0 +1 @@
+#include "../../include/semaphore.h"
