diff --git a/inet/Makefile b/inet/Makefile
index d7139c1..e2dfce4 100644
--- a/inet/Makefile
+++ b/inet/Makefile
@@ -98,3 +98,5 @@ ifeq (yes,$(build-static-nss))
 otherlibs += $(nssobjdir)/libnss_files.a $(resolvobjdir)/libnss_dns.a \
              $(resolvobjdir)/libresolv.a
 endif
+
+CFLAGS-ether_line.c += -O1
diff --git a/stdio-common/Makefile b/stdio-common/Makefile
index f9fc9da..d9bd367 100644
--- a/stdio-common/Makefile
+++ b/stdio-common/Makefile
@@ -127,3 +127,5 @@ bug15-ENV = LOCPATH=$(common-objpfx)localedata
 ifneq (,$(filter %REENTRANT, $(defines)))
 CPPFLAGS += -D_IO_MTSAFE_IO
 endif
+
+CFLAGS-vfscanf.c += -O1
