Index: b/ports/sysdeps/mips/bits/setjmp.h
===================================================================
--- a/ports/sysdeps/mips/bits/setjmp.h	2006-01-10 11:18:16.000000000 +0100
+++ b/ports/sysdeps/mips/bits/setjmp.h	2008-08-08 18:38:02.000000000 +0200
@@ -27,7 +27,7 @@
 
 #include <sgidefs.h>
 
-typedef struct
+typedef struct __mips_jmp_buf
   {
 #if _MIPS_SIM == _ABIO32
     /* Program counter.  */
