diff -Naur eglibc-2.13.orig/localedata/locales/ar_IN eglibc-2.13.new/localedata/locales/ar_IN
--- eglibc-2.13.orig/localedata/locales/ar_IN	2006-08-17 06:48:26.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/ar_IN	2012-05-03 11:28:10.000000000 +0530
@@ -52,7 +52,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U20A8>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3;2
diff -Naur eglibc-2.13.orig/localedata/locales/as_IN eglibc-2.13.new/localedata/locales/as_IN
--- eglibc-2.13.orig/localedata/locales/as_IN	2010-02-04 22:11:35.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/as_IN	2012-05-03 11:28:33.000000000 +0530
@@ -51,7 +51,7 @@
 %%%%%%%%%%%%%
 LC_MONETARY
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U099F><U0995><U09BE>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3;2
diff -Naur eglibc-2.13.orig/localedata/locales/bn_IN eglibc-2.13.new/localedata/locales/bn_IN
--- eglibc-2.13.orig/localedata/locales/bn_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/bn_IN	2012-05-03 11:28:53.000000000 +0530
@@ -53,7 +53,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U099F><U09BE><U0995><U09BE>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          2;3
diff -Naur eglibc-2.13.orig/localedata/locales/en_IN eglibc-2.13.new/localedata/locales/en_IN
--- eglibc-2.13.orig/localedata/locales/en_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/en_IN	2012-05-03 11:29:32.000000000 +0530
@@ -52,7 +52,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U20A8>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3;2
diff -Naur eglibc-2.13.orig/localedata/locales/gu_IN eglibc-2.13.new/localedata/locales/gu_IN
--- eglibc-2.13.orig/localedata/locales/gu_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/gu_IN	2012-05-03 11:31:38.000000000 +0530
@@ -71,7 +71,7 @@
 % for IBM Class for Unicode/Java
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U0AB0><U0AC1>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/hi_IN eglibc-2.13.new/localedata/locales/hi_IN
--- eglibc-2.13.orig/localedata/locales/hi_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/hi_IN	2012-05-03 11:31:58.000000000 +0530
@@ -69,7 +69,7 @@
 % for IBM Class for Unicode/Java
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U0930><U0942>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/kn_IN eglibc-2.13.new/localedata/locales/kn_IN
--- eglibc-2.13.orig/localedata/locales/kn_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/kn_IN	2012-05-03 11:32:35.000000000 +0530
@@ -72,7 +72,7 @@
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
 % *** Change this to reflect Kannada , in Dev it is RA with VS U attached
-currency_symbol       "<U0CB0><U0CC2>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/kok_IN eglibc-2.13.new/localedata/locales/kok_IN
--- eglibc-2.13.orig/localedata/locales/kok_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/kok_IN	2012-05-03 11:32:50.000000000 +0530
@@ -49,7 +49,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U0930><U0941>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/ks_IN eglibc-2.13.new/localedata/locales/ks_IN
--- eglibc-2.13.orig/localedata/locales/ks_IN	2009-04-15 01:16:20.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/ks_IN	2012-05-03 11:33:07.000000000 +0530
@@ -57,7 +57,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U20A8>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/ml_IN eglibc-2.13.new/localedata/locales/ml_IN
--- eglibc-2.13.orig/localedata/locales/ml_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/ml_IN	2012-05-03 11:33:36.000000000 +0530
@@ -69,7 +69,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol     "<U0049><U004E><U0052><U0020>"
-currency_symbol     "<U0D30><U0D42>"
+currency_symbol     "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3;2
diff -Naur eglibc-2.13.orig/localedata/locales/mr_IN eglibc-2.13.new/localedata/locales/mr_IN
--- eglibc-2.13.orig/localedata/locales/mr_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/mr_IN	2012-05-03 11:33:52.000000000 +0530
@@ -70,7 +70,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U0930><U0941>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/or_IN eglibc-2.13.new/localedata/locales/or_IN
--- eglibc-2.13.orig/localedata/locales/or_IN	2009-01-19 23:13:22.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/or_IN	2012-05-03 11:48:34.000000000 +0530
@@ -607,7 +607,7 @@
 %%%%%%%%%%%%%
 LC_MONETARY
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U003D><U0030><U0023><U0052><U0073><U002E><U007C><U0031><U0023><U0052><U0065><U002E><U007C><U0031><U003C><U0052><U0073><U002E>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3;2
diff -Naur eglibc-2.13.orig/localedata/locales/pa_IN eglibc-2.13.new/localedata/locales/pa_IN
--- eglibc-2.13.orig/localedata/locales/pa_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/pa_IN	2012-05-03 11:48:55.000000000 +0530
@@ -66,7 +66,7 @@
 LC_MONETARY
 % This is the POSIX Locale definition the LC_MONETARY category.
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U0A30><U0A41><U0A2A><U0A0F>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/sa_IN eglibc-2.13.new/localedata/locales/sa_IN
--- eglibc-2.13.orig/localedata/locales/sa_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/sa_IN	2012-05-03 11:49:18.000000000 +0530
@@ -52,7 +52,7 @@
 % for IBM Class for Unicode/Java
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U0930><U0942>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/sd_IN eglibc-2.13.new/localedata/locales/sd_IN
--- eglibc-2.13.orig/localedata/locales/sd_IN	2008-12-10 22:09:54.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/sd_IN	2012-05-03 11:49:35.000000000 +0530
@@ -57,7 +57,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U20A8>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/sd_IN@devanagari eglibc-2.13.new/localedata/locales/sd_IN@devanagari
--- eglibc-2.13.orig/localedata/locales/sd_IN@devanagari	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/sd_IN@devanagari	2012-05-03 11:49:50.000000000 +0530
@@ -53,7 +53,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U0930><U0941>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
diff -Naur eglibc-2.13.orig/localedata/locales/ta_IN eglibc-2.13.new/localedata/locales/ta_IN
--- eglibc-2.13.orig/localedata/locales/ta_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/ta_IN	2012-05-03 11:50:11.000000000 +0530
@@ -74,7 +74,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U20A8>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3;2
diff -Naur eglibc-2.13.orig/localedata/locales/te_IN eglibc-2.13.new/localedata/locales/te_IN
--- eglibc-2.13.orig/localedata/locales/te_IN	2010-04-14 20:57:25.000000000 +0530
+++ eglibc-2.13.new/localedata/locales/te_IN	2012-05-03 11:50:30.000000000 +0530
@@ -73,7 +73,7 @@
 % for IBM Class for Unicode.
 %
 int_curr_symbol       "<U0049><U004E><U0052><U0020>"
-currency_symbol       "<U0C30><U0C42><U002E>"
+currency_symbol       "<U20B9>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3;2
