--- glibc-2.6.orig/localedata/locales/am_ET
+++ glibc-2.6/localedata/locales/am_ET
@@ -58,7 +58,7 @@
 %
 %  The ETHIOPIC script declaration must come first.
 %
-script <ETHIOPIC>
+define USE_ETHIOPIC
 %
 % Copy the template from ISO/IEC 14651
 %
--- glibc-2.6.orig/localedata/locales/iso14651_t1_common
+++ glibc-2.6/localedata/locales/iso14651_t1_common
@@ -50,6 +50,9 @@
 # 2005-11-29, Pablo Saratxaga <pablo@mandriva.com>
 
 # Déclaration des systèmes d'écriture / Declaration of scripts
+ifdef USE_ETHIOPIC
+script <ETHIOPIC>
+endif
 script <SPECIAL>
 script <LATIN>
 script <TIFINAGH>
