--- glibc-2.7.orig/localedata/locales/hu_HU
+++ glibc-2.7/localedata/locales/hu_HU
@@ -520,8 +520,8 @@
 t_fmt_ampm "<U0025><U0048><U002E><U0025><U004D><U002E><U0025><U0053>"
 date_fmt "<U0025><U0059><U002E><U0020><U0025><U0062><U002E><U0020><U0025><U0065><U002E><U002C>/
 <U0020><U0025><U0041><U002C><U0020><U0025><U0048><U002E><U0025><U004D><U002E><U0025><U0053><U0020><U0025><U005A>"
-week     7;19971201;4
-first_weekday	1
+%week     7;19971201;4
+first_weekday	2
 first_workday	1
 END LC_TIME
 
