diff -Naur glibc-2.7.orig/localedata/locales/pt_PT glibc-2.7/localedata/locales/pt_PT
--- glibc-2.7.orig/localedata/locales/pt_PT	2006-07-30 23:19:43.000000000 +0100
+++ glibc-2.7/localedata/locales/pt_PT	2008-01-04 15:38:19.000000000 +0000
@@ -65,7 +65,7 @@
 LC_MONETARY
 int_curr_symbol           "<U0045><U0055><U0052><U0020>"
 currency_symbol           "<U20AC>"
-mon_decimal_point         "<U0024>"
+mon_decimal_point         "<U002C>"
 mon_thousands_sep         "<U002E>"
 mon_grouping              3;3
 positive_sign             ""
