Index: glibc-2.3.6/localedata/locales/ml_IN
===================================================================
--- glibc-2.3.6.orig/localedata/locales/ml_IN
+++ glibc-2.3.6/localedata/locales/ml_IN
@@ -104,7 +104,7 @@
 "<U0D2E><U0D3E><U0D30><U0D4D><U200D>";/
 "<U0D0F><U0D2A><U0D4D><U0D30>";/
 "<U0D2E><U0D46>";/
-"<U0D1C><U0D42><U0D7B><U0D23><U0D4D><U200D>";/
+"<U0D1C><U0D42><U0D23><U0D4D>";/
 "<U0D1C><U0D42><U0D32><U0D48>";/
 "<U0D06><U0D17><U0D4D>";/
 "<U0D38><U0D46><U0D2A><U0D4D>";/
Index: glibc-2.3.6/localedata/locales/as_IN
===================================================================
--- glibc-2.3.6.orig/localedata/locales/as_IN
+++ glibc-2.3.6/localedata/locales/as_IN
@@ -596,7 +596,7 @@
 <U09CC> <X689A>;<X0005>;<X0005>;IGNORE
 <U09CD> <X689C>;<X0005>;<X0005>;IGNORE
 <U09D7> <X689E>;<X0005>;<X0005>;IGNORE
-<UNASSIGNED-09CE> "<XE2E9><X6B20>";<X0005>;"<X0005><X00C0>";IGNORE
+%<UNASSIGNED-09CE> "<XE2E9><X6B20>";<X0005>;"<X0005><X00C0>";IGNORE
 UNDEFINED IGNORE;IGNORE;IGNORE;%...
 
 order_end
