Index: b/localedata/Makefile
===================================================================
--- a/localedata/Makefile	2008-03-31 01:40:06.000000000 +0200
+++ b/localedata/Makefile	2008-08-08 18:37:34.000000000 +0200
@@ -223,7 +223,7 @@
 	echo -n `echo $$locale | sed 's/\([^\@]*\)\(\@.*\)*/\2/'`; \
 	echo -n '...'; \
 	input=`echo $$locale | sed 's/\([^.]*\)[^@]*\(.*\)/\1\2/'`; \
-	$(LOCALEDEF) --alias-file=../intl/locale.alias \
+	$(LOCALEDEF) --alias-file=../intl/locale.alias --no-archive \
 		     -i locales/$$input -c -f charmaps/$$charset \
 		     $(addprefix --prefix=,$(install_root)) $$locale; \
 	echo ' done'; \
