Malta joined the Euro zone.

diff --git a/localedata/locales/mt_MT b/localedata/locales/mt_MT
index a882de4..8f0e839 100644
--- a/localedata/locales/mt_MT
+++ b/localedata/locales/mt_MT
@@ -131,8 +131,8 @@ LC_MONETARY
 % These are generated based on XML base Locale defintion file
 % for IBM Class for Unicode.
 %
-int_curr_symbol       "<U004D><U0054><U004C><U0020>"
-currency_symbol       "<U004C><U006D>"
+int_curr_symbol       "<U0045><U0055><U0052><U0020>"
+currency_symbol       "<U20AC>"
 mon_decimal_point     "<U002E>"
 mon_thousands_sep     "<U002C>"
 mon_grouping          3
