http://sourceware.org/bugzilla/show_bug.cgi?id=4557

Index: localedata/locales/as_IN
===================================================================
--- localedata/locales/as_IN.orig
+++ localedata/locales/as_IN
@@ -596,7 +596,7 @@
 <U09CC> <X689A>;<X0005>;<X0005>;IGNORE
 <U09CD> <X689C>;<X0005>;<X0005>;IGNORE
 <U09D7> <X689E>;<X0005>;<X0005>;IGNORE
-<UNASSIGNED-09CE> "<XE2E9><X6B20>";<X0005>;"<X0005><X00C0>";IGNORE
+<U09CE> "<XE2E9><X6B20>";<X0005>;"<X0005><X00C0>";IGNORE
 UNDEFINED IGNORE;IGNORE;IGNORE;%...
 
 order_end
