--- glibc-2.19/localedata/Makefile	2014-02-07 02:04:38.000000000 -0700
+++ glibc-2.19/localedata/Makefile	2014-07-12 17:27:05.063384648 -0600
@@ -292,6 +292,7 @@
 bug-setlocale1-static-ENV = $(bug-setlocale1-ENV)
 bug-setlocale1-static-ARGS = $(bug-setlocale1-ARGS)
 tst-setlocale2-ENV = LOCPATH=$(common-objpfx)localedata
+tst-setlocale3-ENV = LOCPATH=$(common-objpfx)localedata
 
 $(objdir)/iconvdata/gconv-modules:
 	$(MAKE) -C ../iconvdata subdir=iconvdata $@
