# DP: Description: Update to latest CVS
# DP: Date: 2006-09-29

Index: locale/iso-3166.def
===================================================================
--- locale/iso-3166.def.orig
+++ locale/iso-3166.def
@@ -6,6 +6,7 @@
  * http://www.gnu.org/software/libc/bugs.html and follow
  * instructions there to file a bug report.
  */
+DEFINE_COUNTRY_CODE ("AALAND ISLANDS", AX, ALA, 248)
 DEFINE_COUNTRY_CODE ("AFGHANISTAN", AF, AFG, 4)
 DEFINE_COUNTRY_CODE ("ALBANIA", AL, ALB, 8)
 DEFINE_COUNTRY_CODE ("ALGERIA", DZ, DZA, 12)
@@ -149,6 +150,7 @@
 DEFINE_COUNTRY_CODE ("MOLDOVA, REPUBLIC OF", MD, MDA, 498)
 DEFINE_COUNTRY_CODE ("MONACO", MC, MCO, 492)
 DEFINE_COUNTRY_CODE ("MONGOLIA", MN, MNG, 496)
+DEFINE_COUNTRY_CODE ("MONTENEGRO", ME, MNE, 499)
 DEFINE_COUNTRY_CODE ("MONTSERRAT", MS, MSR, 500)
 DEFINE_COUNTRY_CODE ("MOROCCO", MA, MAR, 504)
 DEFINE_COUNTRY_CODE ("MOZAMBIQUE", MZ, MOZ, 508)
