https://sourceware.org/bugzilla/show_bug.cgi?id=16796#c1

Index: eglibc-2.19/ports/sysdeps/aarch64/nptl/tls.h
===================================================================
--- eglibc-2.19.orig/ports/sysdeps/aarch64/nptl/tls.h	2014-01-03 17:51:28.000000000 +0000
+++ eglibc-2.19/ports/sysdeps/aarch64/nptl/tls.h	2150-05-10 02:22:49.085789000 +0000
@@ -62,7 +62,7 @@
 # define TLS_INIT_TCB_SIZE	sizeof (tcbhead_t)
 
 /* Alignment requirements for the initial TCB.  */
-# define TLS_INIT_TCB_ALIGN	__alignof__ (tcbhead_t)
+# define TLS_INIT_TCB_ALIGN	__alignof__ (struct pthread)
 
 /* This is the size of the TCB.  */
 # define TLS_TCB_SIZE		sizeof (tcbhead_t)
@@ -71,7 +71,7 @@
 # define TLS_PRE_TCB_SIZE	sizeof (struct pthread)
 
 /* Alignment requirements for the TCB.  */
-# define TLS_TCB_ALIGN		__alignof__ (tcbhead_t)
+# define TLS_TCB_ALIGN		__alignof__ (struct pthread)
 
 /* Install the dtv pointer.  The pointer passed is to the element with
    index -1 which contain the length.  */
