Index: eglibc-2.17/libio/freopen.c
===================================================================
--- eglibc-2.17.orig/libio/freopen.c
+++ eglibc-2.17/libio/freopen.c
@@ -93,15 +93,16 @@
                       (result->_flags2 & _IO_FLAGS2_CLOEXEC) != 0
                       ? O_CLOEXEC : 0);
 #else
-# define newfd 1
+# define newfd -1
 #endif
 
 #ifndef __ASSUME_DUP3
 	  if (newfd < 0)
 	    {
+# ifdef O_CLOEXEC	    
 	      if (errno == ENOSYS)
 		__have_dup3 = -1;
-
+# endif		
 	      __dup2 (_IO_fileno (result), fd);
 	      if ((result->_flags2 & _IO_FLAGS2_CLOEXEC) != 0)
 		__fcntl (fd, F_SETFD, FD_CLOEXEC);
Index: eglibc-2.17/libio/freopen64.c
===================================================================
--- eglibc-2.17.orig/libio/freopen64.c
+++ eglibc-2.17/libio/freopen64.c
@@ -76,15 +76,16 @@
                       (result->_flags2 & _IO_FLAGS2_CLOEXEC) != 0
                       ? O_CLOEXEC : 0);
 #else
-# define newfd 1
+# define newfd -1
 #endif
 
 #ifndef __ASSUME_DUP3
 	  if (newfd < 0)
 	    {
+# ifdef O_CLOEXEC	    
 	      if (errno == ENOSYS)
 		__have_dup3 = -1;
-
+# endif
 	      __dup2 (_IO_fileno (result), fd);
 	      if ((result->_flags2 & _IO_FLAGS2_CLOEXEC) != 0)
 		__fcntl (fd, F_SETFD, FD_CLOEXEC);
