--- eglibc-2.17.orig/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/i386/linuxthreads/tls.h
+++ eglibc-2.17/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/i386/linuxthreads/tls.h
@@ -21,6 +21,8 @@
 #ifndef _FREEBSD_TLS_H
 #define _FREEBSD_TLS_H
 
+#define HAVE_TLS_SUPPORT 1
+
 #include <linuxthreads/sysdeps/i386/tls.h>
 
 /* We can support TLS only if the floating-stack support is available.
--- eglibc-2.17.orig/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/x86_64/linuxthreads/tls.h
+++ eglibc-2.17/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/x86_64/linuxthreads/tls.h
@@ -21,6 +21,8 @@
 #ifndef _FREEBSD_TLS_H
 #define _FREEBSD_TLS_H
 
+#define HAVE_TLS_SUPPORT 1
+
 #include <linuxthreads/sysdeps/x86_64/tls.h>
 
 #ifdef HAVE_TLS_SUPPORT
--- eglibc-2.17.orig/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/mips/linuxthreads/tls.h
+++ eglibc-2.17/ports/sysdeps/unix/bsd/bsd4.4/kfreebsd/mips/linuxthreads/tls.h
@@ -20,6 +20,8 @@
 #ifndef _TLS_H
 #define _TLS_H
 
+#define HAVE_TLS_SUPPORT 1
+
 #ifndef __ASSEMBLER__
 
 # include <stdbool.h>
